build.sh

```diff
@@ -24,27 +29,76 @@ elif [ $1 = "--help" ] && [ -z "$2" ]; then
   echo ""
   echo "Type '--help name' to find out more about the argument 'name'."
   exit 1
-elif [ $1 = "--help" ] && [ ! -z "$2" ]; then
-  if [ $2 = "port" ]; then
+elif [ "$1" = "--help" ] && [ ! -z "$2" ]; then
+  if [ "$2" = "port" ]; then
     echo "This is the port of the device in which the sketch will be uploaded to. Can be omitted if just verifying."
-  elif [ $2 = "verify" ]; then
+  elif [ "$2"= "verify" ]; then
     echo "Specifying this argument will only compile the script. No uploading will be done."
-  elif [ $2 = "upload" ]; then
+  elif [ "$2" = "upload" ]; then
     echo "Specifying this argument will compile and upload the sketch to the given device port."
   else
     echo "This argument does not exist."
   fi
   exit 1
 else
-  PORT_ARG="--port $1"
-  if [ -z "$2" ]; then
-    COMMAND="--upload"
-  elif [ $2 = "verify" ] || [ $2 = "upload" ]; then
-    COMMAND="--$2"
-  else
-    echo "Invalid argument. Please use --help for more info."
-    exit 1
-  fi   
+  if [ ! "$OPERATING_SYSTEM" = "Linux" ]; then
+    PORT_ARG="--port $1"
+    if [ -z "$2" ]; then
+      COMMAND="--upload"
+    elif [ "$2" = "verify" ] || [ "$2" = "upload" ]; then
+      COMMAND="--$2"
+    else
+      echo "Invalid argument. Please use --help for more info."
+      exit 1
+    fi
+  fi
+fi
+
+# For Linux Users, make sure they have installed libraries and src directory
+if [ "$OPERATING_SYSTEM" = "Linux" ]; then
+  if [ ! -d "./src/" ]; then
+    ln -s Marlin src
+  fi
+  mkdir ./libraries/
+  if [ ! -d "/usr/share/arduino/libraries/LiquidCrystal_I2C" ]; then
+    echo "Missing LiquidCrystal_I2C - Cloning..."
+    cd ./libraries/
+    git clone https://github.com/kiyoshigawa/LiquidCrystal_I2C.git
+    cd $HERE
+    rm -rf ./.build/
```

Comment:
You have to remove the `.build/` directory for ino to search for new plugins that you may have added after running the build command before. I'm just ensuring that if we add a new library, we wipe the build directory automatically.